tests/php/ORM/DataListEagerLoadingTest.php

```diff
@@ -1115,4 +1116,22 @@ public function provideEagerLoadingEmptyRelationLists(): array
             ],
         ];
     }
+
+    public function testFirstHasEagerloadedRelation()
+    {
+        $record = EagerLoadObject::create(['Title' => 'My obj']);
+        $record->write();
+        $record->HasManyEagerLoadObjects()->add(HasManyEagerLoadObject::create(['Title' => 'My related obj']));
+        $obj = EagerLoadObject::get()->eagerLoad('HasManyEagerLoadObjects')->first();
+        $this->assertInstanceOf(ArrayList::class, $obj->HasManyEagerLoadObjects());
```

Comment:
I had originally meant ensure we have x3 objections here to test and ensure that getting ->first() returns the correct one :-) So create another couple of `HasManyEagerLoadObject`'s with different titles